File: facade_pattern/fd.py
class CPU:
  def process(self):
    print("CPU processing...")

class Memory:
  def load(self):
    print("Memory loading...")

class SSD:
  def read(self):
    print("SSD reading...")

class Computer:
  def __init__(self):
    self.cpu = CPU()
    self.memory = Memory()
    self.ssd = SSD()
  # Facade method

  def boost(self):
    print("Computer starting up...")
    self.cpu.process()
    self.memory.load()
    self.ssd.read()
    print("Computer ready to use!")

c = Computer()
c.boost()

File: facade_pattern/fd.cs
using System;

class CPU
{
  public void Process()
  {
    Console.WriteLine("CPU processing...");
  }
}

class Memory
{
  public void Load()
  {
    Console.WriteLine("Memory loading...");
  }
}

class SSD
{
  public void Read()
  {
    Console.WriteLine("SSD reading...");
  }
}

class Computer
{
  private CPU cpu;
  private Memory memory;
  private SSD ssd;

  public Computer()
  {
    cpu = new CPU();
    memory = new Memory();
    ssd = new SSD();
  }

  // Facade method
  public void Boost()
  {
    Console.WriteLine("Computer starting up...");
    cpu.Process();
    memory.Load();
    ssd.Read();
    Console.WriteLine("Computer ready to use!");
  }
}

class Program
{
  static void Main(string[] args)
  {
    Computer c = new Computer();
    c.Boost();
  }
}